### PR TITLE
Add CMake switch for using XCB instead of Xlib for the Vulkan backend

### PR DESCRIPTION
**Pull request by vladfi1**
There may be other places to do this as well.

### CODE REVIEW

**Comment by ligfx** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/VideoBackends/Vulkan/CMakeLists.txt [hunk 1]
```diff
@@ -28,6 +28,10 @@ set(LIBS
   common
 )
 
+if(ENABLE_XCB)
+  list(APPEND LIBS X11-xcb)
```
Comment (Orphis) [inline Source/Core/VideoBackends/Vulkan/CMakeLists.txt]: Add the dependency directly to the Vulkan backend directly instead

File: CMakeLists.txt [hunk 2]
```diff
@@ -544,6 +545,11 @@ if(ENABLE_ANALYTICS)
   add_definitions(-DUSE_ANALYTICS=1)
 endif()
 
+if(ENABLE_XCB)
```
Comment (Orphis) [inline CMakeLists.txt]: Move this to the Vulkan backend CMake file and set the define on the target instead of globally.

**Comment by stenzek** [comment]
IMO there's no need to make this an option, IIRC all modern xlib implementations are wrappers on top of xcb anyway. I just haven't got around to verifying that the XCB implementation in the backend is correct yet.

**Comment by vladfi1** [comment]
@stenzek Ok - XCB seems to work fine for me with this code.

**Pull request closed without merging**